### ISSUE TITLE
Cannot start daemon after upgraded to 1.0.7

### ISSUE THREAD

**Issue by Trypt88** [DogeChia/doge-chia#17]
(venv) doge-chia % dogechia start farmer -r
Error, timeout.
Daemon not started yet
Starting daemon
Traceback (most recent call last):
File "/Users/trypt/doge-chia/venv/bin/dogechia", line 33, in
sys.exit(load_entry_point('dogechia-blockchain', 'console_scripts', 'dogechia')())
File "/Users/trypt/doge-chia/dogechia/cmds/dogechia.py", line 79, in main
cli() # pylint: disable=no-value-for-parameter
File "/Users/trypt/doge-chia/venv/lib/python3.9/site-packages/click/core.py", line 829, in call
return self.main(*args, **kwargs)
File "/Users/trypt/doge-chia/venv/lib/python3.9/site-packages/click/core.py", line 782, in main
rv = self.invoke(ctx)
File "/Users/trypt/doge-chia/venv/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
return _process_result(sub_ctx.command.invoke(sub_ctx))
File "/Users/trypt/doge-chia/venv/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
return ctx.invoke(self.callback, **ctx.params)
File "/Users/trypt/doge-chia/venv/lib/python3.9/site-packages/click/core.py", line 610, in invoke
return callback(*args, **kwargs)
File "/Users/trypt/doge-chia/venv/lib/python3.9/site-packages/click/decorators.py", line 21, in new_func
return f(get_current_context(), *args, **kwargs)
File "/Users/trypt/doge-chia/dogechia/cmds/dogechia.py", line 61, in run_daemon_cmd
asyncio.get_event_loop().run_until_complete(async_run_daemon(ctx.obj["root_path"]))
File "/usr/local/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
return future.result()
File "/Users/trypt/doge-chia/dogechia/daemon/server.py", line 1006, in async_run_daemon
await ws_server.start()
File "/Users/trypt/doge-chia/dogechia/daemon/server.py", line 143, in start
self.websocket_server = await serve(
File "/Users/trypt/doge-chia/venv/lib/python3.9/site-packages/websockets/server.py", line 965, in await_impl
server = await self._create_server()
File "/usr/local/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/lib/python3.9/asyncio/base_events.py", line 1494, in create_server
raise OSError(err.errno, 'error while attempting '
OSError: [Errno 48] error while attempting to bind on address ('::1', 56669, 0, 0): address already in use
Error, timeout.
Daemon not started yet
Failed to create the dogechia daemon
(venv) doge-chia % dogechia stop all -d
Error, timeout.
Daemon not started yet
Couldn't connect to dogechia daemon

I'm on Mac, after upgrading to 1.0.7 via terminal and activated, trying to start but failed, cannot even stop all, all gives me error, timeout. After several tries, I gave up, paste the old 0.1.5 back to the folder, all normal again.

**Comment by Jsewill** [comment]
I'm seeing the same on linux, with the following from the debug.log:

```
2021-07-27T17:28:06.649 daemon dogechia.daemon.server     : INFO     Websocket exception. Closing websocket with Unknown code = 1006 (connection closed abnormally [internal]), no reason Traceback (most recent call last):
  File "/home/farmer/Crypto/dogechia/dogechia-blockchain/venv/lib/python3.7/site-packages/websockets/protocol.py", line 827, in transfer_data
    message = await self.read_message()
  File "/home/farmer/Crypto/dogechia/dogechia-blockchain/venv/lib/python3.7/site-packages/websockets/protocol.py", line 895, in read_message
    frame = await self.read_data_frame(max_size=self.max_size)
  File "/home/farmer/Crypto/dogechia/dogechia-blockchain/venv/lib/python3.7/site-packages/websockets/protocol.py", line 971, in read_data_frame
    frame = await self.read_frame(max_size)
  File "/home/farmer/Crypto/dogechia/dogechia-blockchain/venv/lib/python3.7/site-packages/websockets/protocol.py", line 1051, in read_frame
    extensions=self.extensions,
  File "/home/farmer/Crypto/dogechia/dogechia-blockchain/venv/lib/python3.7/site-packages/websockets/framing.py", line 105, in read
    data = await reader(2)
  File "/usr/lib/python3.7/asyncio/streams.py", line 677, in readexactly
    raise IncompleteReadError(incomplete, n)
asyncio.streams.IncompleteReadError: 0 bytes read on a total of 2 expected bytes

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/farmer/Crypto/dogechia/dogechia-blockchain/dogechia/daemon/server.py", line 172, in safe_handle
    async for message in websocket:
  File "/home/farmer/Crypto/dogechia/dogechia-blockchain/venv/lib/python3.7/site-packages/websockets/protocol.py", line 439, in __aiter__
    yield await self.recv()
  File "/home/farmer/Crypto/dogechia/dogechia-blockchain/venv/lib/python3.7/site-packages/websockets/protocol.py", line 509, in recv
    await self.ensure_open()
  File "/home/farmer/Crypto/dogechia/dogechia-blockchain/venv/lib/python3.7/site-packages/websockets/protocol.py", line 803, in ensure_open
    raise self.connection_closed_exc()
websockets.exceptions.ConnectionClosedError: code = 1006 (connection closed abnormally [internal]), no reason

```

**Comment by zahid-a** [comment]
v1.0.8 daemon not starting

**Comment by enterovipunen** [comment]
OSError: [Errno 48] error while attempting to bind on address ('::1', 56669, 0, 0): address already in use

That typically means that the process not shut down completely before install. 

Ensure processes down before attempting install.
`dogechia stop all`

**Comment by Ilmairinen** [comment]
On Ubuntu 20.04
dogechia stop all
ps -ef | grep dogechia
(I didn't have anything come back, but kill anything still running.)
cd
rm -rf doge-chia
git clone --depth 1 --branch 1.0.8 https://github.com/DogeChia/doge-chia.git
cd doge-chia
sh install.sh
. ./activate
dogechia init
sh install-gui.sh
cd dogechia-blockchain-gui
npm run electron &

Gui starts up without error.

From CLI:
dogechia version
returns: 1.0.8

I couldn't reproduce the bug you reported.

**Comment by Qwinn1** [comment]
I can reproduce the error, and not only that, I get it in Windows as well.

The GUI starts fine, at least on Ubuntu 20.04.  The daemon failure occurs when I try to do "dogechia start harvester -r" on any of my harvesters, Ubuntu or Windows.  I did set the port to my farmer in my farmer_peer section of my harvester config.yamls, and I did set up the certificates properly (already experienced doing it for chia and flax).

**Comment by Qwinn1** [comment]
(I will note that I didn't bother installing the GUI on any of my harvesters, since I shouldn't need it there).

**Comment by Qwinn1** [comment]
Additional info:  Despite it saying that it failed to create the dogechia daemon, doing a ps -ef | grep dogechia DOES in fact return a running dogechia_daemon process.  Killing it and then attempting to start the harvesters doesn't help matters, it just claims it can't start the daemon and yet leaves a running daemon process anyway.

**Comment by Trypt88** [comment]
Hi, so I've followed your instructions, also I did exactly on 1.0.8 steps told, still... daemon won't start, it freezes for like 15 seconds, then it returned:
Error, timeout.
Daemon not started yet
Failed to create the dogechia daemon

And below is what I got in debug.log:
2021-07-29T17:09:53.321 daemon dogechia.daemon.server     : INFO     Starting Daemon Server
2021-07-29T17:09:53.325 daemon dogechia.daemon.server     : INFO     Waiting Daemon WebSocketServer closure
2021-07-29T17:10:25.409 daemon dogechia.daemon.server     : INFO     Websocket exception. Closing websocket with Unknown code = 1006 (connection closed abnormally [internal]), no reason Traceback (most recent call last):
  File "/Users/trypt/doge-chia/venv/lib/python3.9/site-packages/websockets/protocol.py", line 827, in transfer_data
    message = await self.read_message()
  File "/Users/trypt/doge-chia/venv/lib/python3.9/site-packages/websockets/protocol.py", line 895, in read_message
    frame = await self.read_data_frame(max_size=self.max_size)
  File "/Users/trypt/doge-chia/venv/lib/python3.9/site-packages/websockets/protocol.py", line 971, in read_data_frame
    frame = await self.read_frame(max_size)
  File "/Users/trypt/doge-chia/venv/lib/python3.9/site-packages/websockets/protocol.py", line 1047, in read_frame
    frame = await Frame.read(
  File "/Users/trypt/doge-chia/venv/lib/python3.9/site-packages/websockets/framing.py", line 105, in read
    data = await reader(2)
  File "/usr/local/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/lib/python3.9/asyncio/streams.py", line 721, in readexactly
    raise exceptions.IncompleteReadError(incomplete, n)
asyncio.exceptions.IncompleteReadError: 0 bytes read on a total of 2 expected bytes

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/trypt/doge-chia/dogechia/daemon/server.py", line 172, in safe_handle
    async for message in websocket:
  File "/Users/trypt/doge-chia/venv/lib/python3.9/site-packages/websockets/protocol.py", line 439, in __aiter__
    yield await self.recv()
  File "/Users/trypt/doge-chia/venv/lib/python3.9/site-packages/websockets/protocol.py", line 509, in recv
    await self.ensure_open()
  File "/Users/trypt/doge-chia/venv/lib/python3.9/site-packages/websockets/protocol.py", line 803, in ensure_open
    raise self.connection_closed_exc()
websockets.exceptions.ConnectionClosedError: code = 1006 (connection closed abnormally [internal]), no reason

Very frustrated...

**Comment by Qwinn1** [comment]
Just to let folks know... the 1.0.8 GUI farmer works just fine.  To get past this issue for the time being I just wound up accessing the bulk of my plots via network shares instead of using the harvester model.  Works fine that way, got my first 2 XDG already (just started farming last night with 3878 plots, so 4 hour ETW).  I will just leave it like that until the harvester issue gets resolved.  Slightly slower response times is better than none at all!

**Comment by enterovipunen** [comment]
Looking into it. Marked 1.0.7 and 1.0.8 as prerelease to fix

**Comment by Qwinn1** [comment]
Update:  I downgraded all of my harvesters to version 0.1.5.  Running "dogechia start harvester -r" on them after that worked perfectly well, and they all seem to be successfully communicating with my farmer (which is still 1.0.8) now.

Let me know if any problem can be foreseen in leaving my farmer at version 1.0.8 with 0.1.5 harvesters until the next release, otherwise I'll just leave it as it is.

**Comment by enterovipunen** [comment]
Errant PR from chia was merged in 1.0.7. Reverted for 1.0.9 shold be out today

**Comment by Qwinn1** [comment]
Can confirm 1.0.9 fixes the issues for me on both Ubuntu and Windows.  Thanks!

**Comment by Ilmairinen** [comment]
Anyone objection to closing this issue as resolved?

**Comment by Trypt88** [comment]
Thanks! I will trying 1.0.9 now, gimme a moment~

**Comment by Trypt88** [comment]
The issue is resolved! 1.0.9 runs smoothly, thank you!

**Comment by zahid-a** [comment]
confirm, 1.0.9 works fine on ubuntu 20.04, same on win10
